### ISSUE TITLE
Tab completion inserts text in the wrong place

### ISSUE THREAD

**Issue by seanfarley** [xonsh/xonsh#1635]
When trying to tab-complete a variable, everything works fine:

```
$ $FOO = "~/sandbox"
$ ls $FO<TAB>
```

but trying to expand a path breaks `$ ls $FOO/<tab>` turns into:

```
$ ls $FOO//Users/sean/sandbox/
```

**Comment by gforsyth** [comment]
Hey @seanfarley -- thanks for reporting!  Could you post the output of `xonfig`?  Thanks!

**Comment by mitnk** [comment]
On Mac, could reproduce this issue with PTK shell. Readline shell has no such issue.

**Comment by seanfarley** [comment]
Hmmm, did my last comment not go through? Here is the output of `xonfig`:

```
$ xonfig
+------------------+-----------------+
| xonsh            | 0.4.5.dev114    |
| Git SHA          | a89c612         |
| Python           | 3.5.2           |
| PLY              | 3.7             |
| have readline    | True            |
| prompt toolkit   | 1.0.3           |
| shell type       | prompt_toolkit  |
| pygments         | 2.1.3           |
| on posix         | True            |
| on linux         | False           |
| on darwin        | True            |
| on windows       | False           |
| on cygwin        | False           |
| is superuser     | False           |
| default encoding | utf-8           |
| xonsh encoding   | utf-8           |
| encoding errors  | surrogateescape |
+------------------+-----------------+
```

**Comment by seanfarley** [comment]
I can confirm with @mitnk observed: `readline` appears to have no issue.

**Comment by gforsyth** [comment]
Sorry guys -- I introduced this in #1630 when trying to deal with prompt_toolkit attribute completion.

**Comment by gforsyth** [comment]
Hey @mitnk and @seanfarley -- potential fix in #1639

**Comment by seanfarley** [comment]
Yep, that fixes it for me! Thanks!
